### DISCUSSION TITLE
Structs with reference fields

### DISCUSSION THREAD

**Discussion by felixf4xu** [vlang/v#18541] [Questions and Answers] [answered]
Hi,

I'm reading the document of https://github.com/vlang/v/blob/master/doc/docs.md#structs-with-reference-fields, it says
>Structs with references require explicitly setting the initial value to a reference value unless the struct already defines its own initial value.
Zero-value references, or nil pointers, will NOT be supported in the future

I'm fine with the language design, but I'd like to ask what's the suggested solution if I can't use nil to initialize a reference in a structure?

```
struct Node {
	a &Node
	b &Node = unsafe { nil } // Auto-initialized to nil, use with caution!
}
```
I don't like the `unsafe` keyword here, if I'm writing a purely `v` code without `c` interop, the word of `unsafe ` should be avoided as much as possible, from language design point of view.
```
struct Node {
	a &Node
	b &Node = 0
}
```
What will be the designed solution in the future?

**Comment by Yaisis** [comment, marked as answer]
> What will be the designed solution in the future?

```v
struct Node {
	a ?&Node
	b ?&Node
}

n := Node{}
c := n.a or { ... }
```
